pkg/rpaas/client/client.go

```diff
@@ -153,6 +153,42 @@ type UpdateCertManagerArgs struct {
 	Instance string
 }
 
+type ListUpstreamOptionsArgs struct {
+	Instance string
+}
+
+type AddUpstreamOptionsArgs struct {
+	Instance             string
+	PrimaryBind          string
+	CanaryBinds          []string
+	TrafficShapingPolicy TrafficShapingPolicy
+	LoadBalance          string
+	LoadBalanceHashKey   string
+}
+
+type UpdateUpstreamOptionsArgs struct {
+	Instance             string
+	PrimaryBind          string
+	CanaryBinds          []string
+	TrafficShapingPolicy TrafficShapingPolicy
+	LoadBalance          string
+	LoadBalanceHashKey   string
+}
+
+type TrafficShapingPolicy struct {
```

Comment:
If this is the same struct as defined [here](https://github.com/tsuru/rpaas-operator/pull/182/files#diff-b23056cbe162e26491cb080630ac168d04f4e5517992cc8312488d5f241b1a7cR253), maybe only reference the other definition?